src/actions/ExchangeRateActions.ts

```diff
@@ -238,18 +247,45 @@ async function fetchExchangeRates(account: EdgeAccount, accountIsoFiat: string,
 /**
  * Converts rates from the cache format to the GUI's in-memory format.
  */
-function buildGuiRates(rateCache: ExchangeRateCache): GuiExchangeRates {
+function buildGuiRates(rateCache: ExchangeRateCache, yesterday: string): { exchangeRates: GuiExchangeRates; exchangeRatesMap: GuiExchangeRatesMap } {
   const out: GuiExchangeRates = { 'iso:USD_iso:USD': 1 }
+  const outMap: GuiExchangeRatesMap = new Map()
+  const yesterdayTimestamp = Date.parse(yesterday)
+
   for (const key of Object.keys(rateCache)) {
     const { rate } = rateCache[key]
     out[key] = rate
 
     // Include reverse rates:
-    const codes = key.split('_')
-    const reverseKey = `${codes[1]}_${codes[0]}${codes[2] ? '_' + codes[2] : ''}`
+    const [codeA, codeB, date] = key.split('_') as [string, string, string | undefined]
+    const reverseKey = `${codeB}_${codeA}${date ? '_' + date : ''}`
     out[reverseKey] = rate === 0 ? 0 : 1 / rate
+
+    // Set up exchange rate map. This nest map is keyed This map will hold current rate and 24 hour rate, if available.
+    if (outMap.get(codeA)?.get(codeB) != null || date != null) {
+      continue
+    }
+
+    let yesterdayRate: number | undefined
+    // We only look up yesterday's rate for USD pairs
+    if (codeB === 'iso:USD') {
+      yesterdayRate = rateCache[`${codeA}_${codeB}_${yesterday}`]?.rate ?? closestRateForTimestamp(rateCache, codeA, yesterdayTimestamp)
+    }
+
+    const codeAMap = outMap.get(codeA) ?? new Map()
+    outMap.set(codeA, codeAMap.set(codeB, { currentRate: rate, yesterdayRate }))
+
+    const codeBMap = outMap.get(codeB) ?? new Map()
+    outMap.set(
+      codeB,
+      codeBMap.set(codeA, { currentRate: out[reverseKey], yesterdayRate: yesterdayRate === 0 || yesterdayRate == null ? 0 : 1 / yesterdayRate })
+    )
```

Comment:
Oh, interesting. I see that we are keeping the reverse rate thing going here. I was under the impression were going to always have the outer map be the crypto, and the inner map be the fiat, and therefore get rid of the need for reverse rates, since we always know which way is which.

Reply:
I started with this assumption but there are places in the app that use crypto/crypto rates. Not sure where fiat/crypto rates would be used, but I ultimately figured it would be safer to just mirror the current rates object for now